RecoLocalCalo/HGCalRecProducers/interface/HGCalClusteringAlgoBase.h

```diff
@@ -0,0 +1,97 @@
+#ifndef RecoLocalCalo_HGCalRecProducers_HGCalClusteringAlgoBase_h
+#define RecoLocalCalo_HGCalRecProducers_HGCalClusteringAlgoBase_h
+
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "DataFormats/HGCRecHit/interface/HGCRecHitCollections.h"
+#include "DataFormats/Math/interface/Point3D.h"
+#include "DataFormats/EgammaReco/interface/BasicCluster.h"
+
+#include "RecoLocalCalo/HGCalRecAlgos/interface/RecHitTools.h"
+
+// C/C++ headers
+#include <vector>
+#include <numeric>
+
+namespace hgcal_clustering {
+template <typename T>
+std::vector<size_t> sorted_indices(const std::vector<T> &v) {
```

Comment:
this pattern is repeated in `sort_by_delta` in `HGCalCLUEAlgo.h`. It could be useful to define a more generic version to be used anywhere:
```
template <typename T, typename L>
std::vector<size_t> sorted_indices(const std::vector<T> &v, L sort_func);
```

`std::function` could be used rather than a second template argument for the lambda w/ capture, but it tends to be slower.